Makefile

```diff
@@ -46,16 +46,18 @@ help: ## Display this help.
 # Targets that depend on .gits-commit can use $(shell cat .git-commit) to get a
 # git revision string.  They will only be rebuilt if the revision string
 # actually changes.
-# TODO When a release is created change these steps to use: git describe --abbrev=0 --tags to pull the latest release tag on a specific branch.
 .PHONY: .git-commit.tmp
 .git-commit: .git-commit.tmp
 	@cmp $< $@ >/dev/null 2>&1 || cp $< $@
 .git-commit.tmp:
-	@printf "$$(git rev-parse HEAD 2>/dev/null)" >$@
+	@printf "$$(git rev-parse HEAD 2>/dev/null || echo unknown)" >$@
 	@test -n "$$(git status --porcelain --untracked-files=no)" && echo -dirty >>$@ || true
 
-GOFLAGS += -ldflags="-X 'github.com/confidential-containers/cloud-api-adaptor/cmd.VERSION=$(shell echo "unknown")' \
-                     -X 'github.com/confidential-containers/cloud-api-adaptor/cmd.COMMIT=$(shell cat .git-commit)'"
+version = $(shell git describe --match "v[0-9]*" --abbrev=0 --tags --dirty=-dev || echo unknown)
```

Comment:
Do we want the response to be unknown against a branch? 


For example, the version branch produces this: 
`git describe --match "v[0-9]*" --abbrev=0 --tags --dirty=-dev`
```
fatal: No names found, cannot describe anything.
```

Reply:
@Jordan9500 Thank you very much for pointing out the problem. I was able to reproduce the problem by directly cloning my personal repo. I didn't noticed the problem, since I am usually working in a local repository that is cloned from this upstream repo.

I don't have a good idea to work around the problem when tags are not available. Possibly, we can do something when versioning system for dependencies is introduced #598. Please let me proceed with the current approach for now.